tests/admin/views/test_releases_v2.py

```diff
@@ -0,0 +1,1173 @@
+            old_data_versions["platforms"][pname]["locales"][lname] = 1
+            new_data_versions["platforms"][pname]["locales"][lname] = 2
+
+    ret = api.put("/v2/releases/Firefox-60.0b3-build1", json={"blob": firefox_60_0b3_build1, "product": "Firefox", "old_data_versions": old_data_versions})
+    assert ret.status_code == 400, ret.data
+    assert "Cannot overwrite" in ret.json["exception"]
+
+
+@pytest.mark.usefixtures("releases_db", "mock_verified_userinfo")
+def test_overwrite_succeeds(api, firefox_60_0b3_build1):
+    firefox_60_0b3_build1 = deepcopy(firefox_60_0b3_build1)
+    firefox_60_0b3_build1["detailsUrl"] = "https://newurl"
+    firefox_60_0b3_build1["platforms"]["Linux_x86_64-gcc3"]["locales"]["en-US"]["buildID"] = "9999999999999"
+
+    old_data_versions = infinite_defaultdict()
+    old_data_versions["."] = 1
+    old_data_versions["platforms"]["Linux_x86_64-gcc3"]["locales"]["en-US"] = 1
+    new_data_versions = infinite_defaultdict()
+    new_data_versions["."] = 2
+    new_data_versions["platforms"]["Linux_x86_64-gcc3"]["locales"]["en-US"] = 2
+
+    ret = api.put("/v2/releases/Firefox-60.0b3-build1", json={"blob": firefox_60_0b3_build1, "product": "Firefox", "old_data_versions": old_data_versions})
+    assert ret.status_code == 200, ret.data
+    assert ret.json == new_data_versions
+
+    base_blob = dbo.releases_json.t.select().where(dbo.releases_json.name == "Firefox-60.0b3-build1").execute().fetchone().data
+    locale_blob = (
+        dbo.release_assets.t.select()
+        .where(dbo.release_assets.name == "Firefox-60.0b3-build1")
+        .where(dbo.release_assets.path == ".platforms.Linux_x86_64-gcc3.locales.en-US")
+        .execute()
+        .fetchone()
+        .data
+    )
+    assert base_blob["detailsUrl"] == "https://newurl"
+    assert locale_blob["buildID"] == "9999999999999"
+    assert "locales" not in base_blob["platforms"]["Linux_x86_64-gcc3"]
+    base_history = get_release_history("Firefox-60.0b3-build1")
+    assert len(base_history) == 3
+    # Check to make sure "changed_by" was set
+    assert "-bob.json" in base_history[2][0]
+    locale_history = get_release_assets_history("Firefox-60.0b3-build1", ".platforms.Linux_x86_64-gcc3.locales.en-US")
+    assert len(locale_history) == 3
+    assert "-bob.json" in locale_history[2][0]
+
+    unchanged_locale = (
+        dbo.release_assets.t.select()
+        .where(dbo.release_assets.name == "Firefox-60.0b3-build1")
+        .where(dbo.release_assets.path == ".platforms.Linux_x86_64-gcc3.locales.af")
+        .execute()
+        .fetchone()
+    )
+    assert unchanged_locale["data_version"] == 1
+    unchanged_locale_history = get_release_assets_history("Firefox-60.0b3-build1", ".platforms.Linux_x86_64-gcc3.locales.af")
+    assert len(unchanged_locale_history) == 2
+
+
+@pytest.mark.usefixtures("releases_db", "mock_verified_userinfo")
+def test_overwrite_succeeds_for_new_release(api, firefox_62_0_build1):
```

Comment:
I can see why it's somewhat confusing. Overwrite doesn't obviously mean "replace existing object or create it if it doesn't exist".

Reply:
I think s/overwrite/insert/ might be confusing in a different way, since the endpoint and tests handle updates and deletes too. What do you think about s/overwrite/set/?

Reply:
Sounds good.